### PR TITLE
Regression scripts for OnWayPointChange

### PR DESCRIPTION
**Pull request by GetmanetsIrina**
- boundary values by params
 - out of bound values by params
 - invalid characters \t, \n, whitespace only
 - invalid parameter types
 - invalid json
 - processing API in different HMI levels

### CODE REVIEW

File: test_scripts/API/Navigation/OnWayPointChange/013_OnWayPointChange_invalid_json.lua [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/API/Navigation/commonNavigation')
+
+--[[ Local Functions ]]
+local function onWayPointChange(self)
+  -- missed :
+  local text = '{"params":{"wayPoints":[{"locationName""Hotel"}]},"method":"Navigation.OnWayPointChange","jsonrpc":"2.0"}'
+  self.hmiConnection:Send(text)
+  -- self.hmiConnection:SendNotification("Navigation.OnWayPointChange", notification)
```
Comment (dboltovskyi) [inline test_scripts/API/Navigation/OnWayPointChange/013_OnWayPointChange_invalid_json.lua]: @GetmanetsIrina Please remove commented code.

File: test_scripts/API/Navigation/OnWayPointChange/014_OnWayPointChange_WithInvalidJsonStructure.lua [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+-- Requirement summary:
+-- [OnWayPointChange] As a mobile application I want to be able to be notified on changes
+-- to Destination or Waypoints based on my subscription
+--
+-- Description:
+-- In case:
+-- 1) SDL and HMI are started, Navi interface and embedded navigation source are available on HMI,
+--    mobile applications are registered on SDL and subscribed on destination and waypoints changes notification
+-- 2) Any change in destination or waypoints is registered on HMI (user set new route, canselled the route,
+--    arrived at destination point or crossed a waypoint)
```
Comment (dboltovskyi) [inline test_scripts/API/Navigation/OnWayPointChange/014_OnWayPointChange_WithInvalidJsonStructure.lua]: @GetmanetsIrina Please update description according to script

File: test_scripts/API/Navigation/OnWayPointChange/017_OnWayPointChange_SeveralNotificationsWithIdenticalParams_AllSuccess.lua [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+-- Requirement summary:
+-- [OnWayPointChange] As a mobile application I want to be able to be notified on changes
+-- to Destination or Waypoints based on my subscription
+--
+-- Description:
+-- In case:
+-- 1) SDL and HMI are started, Navi interface and embedded navigation source are available on HMI,
+--    mobile applications are registered on SDL and subscribed on destination and waypoints changes notification
+-- 2) Any change in destination or waypoints is registered on HMI (user set new route, canselled the route,
+--    arrived at destination point or crossed a waypoint) and HMI sends several fake params
```
Comment (dboltovskyi) [inline test_scripts/API/Navigation/OnWayPointChange/017_OnWayPointChange_SeveralNotificationsWithIdenticalParams_AllSuccess.lua]: @GetmanetsIrina Please update description according to script

File: test_scripts/API/Navigation/OnWayPointChange/017_OnWayPointChange_SeveralNotificationsWithIdenticalParams_AllSuccess.lua [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+-- SDL must:
+-- 1) Transfer the notification about changes to destination or waypoints to mobile application without fake params
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/API/Navigation/commonNavigation')
+local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
```
Comment (dboltovskyi) [inline test_scripts/API/Navigation/OnWayPointChange/017_OnWayPointChange_SeveralNotificationsWithIdenticalParams_AllSuccess.lua]: @GetmanetsIrina Module 'commonFunctions' is unused

File: test_scripts/API/Navigation/OnWayPointChange/018_OnWayPointChange_SeveralNotificationsWithDiffParams_AllSuccess.lua [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+-- Requirement summary:
+-- [OnWayPointChange] As a mobile application I want to be able to be notified on changes
+-- to Destination or Waypoints based on my subscription
+--
+-- Description:
+-- In case:
+-- 1) SDL and HMI are started, Navi interface and embedded navigation source are available on HMI,
+--    mobile applications are registered on SDL and subscribed on destination and waypoints changes notification
+-- 2) Any change in destination or waypoints is registered on HMI (user set new route, canselled the route,
+--    arrived at destination point or crossed a waypoint) and HMI sends several fake params
```
Comment (dboltovskyi) [inline test_scripts/API/Navigation/OnWayPointChange/018_OnWayPointChange_SeveralNotificationsWithDiffParams_AllSuccess.lua]: @GetmanetsIrina Please update description according to script

File: test_scripts/API/Navigation/OnWayPointChange/013_OnWayPointChange_invalid_json.lua (hunk 1 above)
Comment (GetmanetsIrina) [reply test_scripts/API/Navigation/OnWayPointChange/013_OnWayPointChange_invalid_json.lua]: Done

File: test_scripts/API/Navigation/OnWayPointChange/014_OnWayPointChange_WithInvalidJsonStructure.lua (hunk 2 above)
Comment (GetmanetsIrina) [reply test_scripts/API/Navigation/OnWayPointChange/014_OnWayPointChange_WithInvalidJsonStructure.lua]: Done

File: test_scripts/API/Navigation/OnWayPointChange/017_OnWayPointChange_SeveralNotificationsWithIdenticalParams_AllSuccess.lua (hunk 3 above)
Comment (GetmanetsIrina) [reply test_scripts/API/Navigation/OnWayPointChange/017_OnWayPointChange_SeveralNotificationsWithIdenticalParams_AllSuccess.lua]: Done

File: test_scripts/API/Navigation/OnWayPointChange/017_OnWayPointChange_SeveralNotificationsWithIdenticalParams_AllSuccess.lua (hunk 4 above)
Comment (GetmanetsIrina) [reply test_scripts/API/Navigation/OnWayPointChange/017_OnWayPointChange_SeveralNotificationsWithIdenticalParams_AllSuccess.lua]: Removed

File: test_scripts/API/Navigation/OnWayPointChange/018_OnWayPointChange_SeveralNotificationsWithDiffParams_AllSuccess.lua (hunk 5 above)
Comment (GetmanetsIrina) [reply test_scripts/API/Navigation/OnWayPointChange/018_OnWayPointChange_SeveralNotificationsWithDiffParams_AllSuccess.lua]: Done

**Pull request merged**